rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -432,6 +413,23 @@ impl LightFetch {
 			Either::B(extract_transaction)
 		}))
 	}
+
+	fn send_requests<T, F>(&self, reqs: Vec<OnDemandRequest>, parse_response: F) -> BoxFuture<T> where
```

Comment:
you might be able to make this more generic by accepting a `R: on_demand::RequestAdapter` and `F: FnOnce(R::Out) -> T`.

Note that this trait is implemented for `Vec<OnDemandRequest>`
A high level API would help you construct tuples with consistent back-references and then we actually save a few allocations on each request.